Sources/SwiftProtobuf/_MessageStorage+BinaryDecoding.swift

```diff
@@ -234,10 +234,17 @@ extension _MessageStorage {
     }
 
     /// Decodes the next field in the reader as an unknown field.
-    private func decodeUnknownField(from reader: inout WireFormatReader, tag: FieldTag) throws {
-        // TODO: Store unknown fields in the unknown storage blob. For now, we just skip them by
-        // reading and discarding them, which is fine to get some basic functionality working but
-        // less efficient than we'll want our final implementation to be.
-        _ = try reader.sliceBySkippingField(tag: tag)
+    ///
+    /// - Parameters:
+    ///   - reader: The `WireFormatReader` from which to read the next field.
+    ///   - tag: The tag representing the current field that was just read from the reader.
+    ///   - discard: If true, the field's data should be skipped. Otherwise, it will be stored in
+    ///     the unknown fields storage.
+    /// - Throws: `BinaryDecodingError` if an error occurred while reading from the buffer.
+    private func decodeUnknownField(from reader: inout WireFormatReader, tag: FieldTag, discard: Bool) throws {
+        let slice = try reader.sliceBySkippingField(tag: tag)
+        if !discard, let pointer = slice.baseAddress, slice.count > 0 {
+            unknownFields.append(protobufData: Data(bytes: pointer, count: slice.count))
```

Comment:
Since I think `append` is already an internal api, it might make sense to have a version that can directly take the slice and skip making a `Data` from it.

Reply:
Done.